### DISCUSSION TITLE
[javascript deep dive] 2장. 자바스크립트란?

### DISCUSSION THREAD

**Discussion by rladpwl0512** [woowacourse-study/2022-javascript-bank#1] [General]
[javascript deep dive] 2장. 자바스크립트란?

**Comment by jin7969** [comment]
자바스크립트는 왜 탄생되었나요?

**Reply by jin7969** [reply]
1. 브라우저에서 동작하기 위해서
2. 동적인 작업을 하기 위해서

**Comment by jin7969** [comment]
크로스 브라우징 이란?

**Reply by jin7969** [reply]
어떤 브라우저를 쓰더라도 동일하게 동작해야 하는 것

**Comment by jin7969** [comment]
컴파일러 언어와 인터프리터 언어 차이

**Reply by jin7969** [reply]
<img width="552" alt="스크린샷 2022-09-13 오후 7 44 04" src="https://user-images.githubusercontent.com/82227098/189881630-4b2607a6-1b9f-4baa-b018-d1764b70cb8b.png">

**Comment by jswith** [comment]
자바스크립트 엔진의 동작 방식에 대해 설명해주세요

**Reply by jswith** [reply]
https://joshua1988.github.io/web-development/translation/javascript/how-js-works-inside-engine/

**Comment by jswith** [comment]
자바스크립트는 싱글스레드인데 왜 비동기가 가능할까요?

**Reply by jswith** [reply]
- 싱글 스레드 : 한 번에 하나의 일
- 동기 : 순서대로
- 비동기 : 순서 보장 X
- 키워드 : 이벤트 루프와 콜백 큐

**Comment by rladpwl0512** [comment]
자바스크립트가 표준화가 된 배경은 무엇일까요? (크로스 브라우징)

**Reply by rladpwl0512** [reply]
크로스 브라우징 이슈를 해결하기 위해. 
초반 javascript는 표준화가 되지 않아서, 브라우저마다 동작되는 것이 있고 동작되지 않는 것이 생겼다. 
따라서 이를 해결하기 위해 javascript를 표준화했다.
